src/main/java/com/tetrahedron/app/DemoProject.java

```diff
@@ -0,0 +1,199 @@
+                    group.translateYProperty().set(group.getTranslateY() + 1);
+                    break;
+                case W:
+                    group.translateYProperty().set(group.getTranslateY() - 1);
+                    break;
+                case D:
+                    group.translateXProperty().set(group.getTranslateX() + 1);
+                    break;
+                case A:
+                    group.translateXProperty().set(group.getTranslateX() - 1);
+                    break;
+                case UP:
+                    group.rotateByX(10);
+                    break;
+                case DOWN:
+                    group.rotateByX(-10);
+                    break;
+                case RIGHT:
+                    group.rotateByY(10);
+                    break;
+                case LEFT:
+                    group.rotateByY(-10);
+                    break;
+            }
+        });
+
+        primaryStage.setScene(scene);
+    }
+
+    public MeshView getMeshView() {
+        TriangleMesh mesh = new TriangleMesh();
+        mesh.getPoints().addAll(vertices);
+        mesh.getTexCoords().addAll(texture);
+        mesh.getFaces().addAll(faces);
+
+        MeshView meshView = new MeshView();
+        meshView.setMesh(mesh);
+        meshView.setMaterial(new PhongMaterial(Color.YELLOW));
+        meshView.setCullFace(CullFace.NONE);
+
+        return meshView;
+    }
+
+    private void initMouseControl(SmartGroup group, Scene scene, Stage stage) {
+        Rotate xRotate;
+        Rotate yRotate;
+        group.getTransforms().addAll(
+                xRotate = new Rotate(0, Rotate.X_AXIS),
+                yRotate = new Rotate(0, Rotate.Y_AXIS)
+        );
+        xRotate.angleProperty().bind(angleX);
+        yRotate.angleProperty().bind(angleY);
+
+        scene.setOnMousePressed(event -> {
+            anchorX = event.getSceneX();
+            anchorY = event.getSceneY();
+            anchorAngleX = angleX.get();
+            anchorAngleY = angleY.get();
+        });
+
+        scene.setOnMouseDragged(event -> {
+            angleX.set(anchorAngleX - (anchorY - event.getSceneY()));
+            angleY.set(anchorAngleY + anchorX - event.getSceneX());
+        });
+
+        stage.addEventHandler(ScrollEvent.SCROLL, event -> {
+            double delta = event.getDeltaY();
+            group.translateZProperty().set(group.getTranslateZ() + delta);
+        });
+    }
+
+    private void loadVertices() {
+        vertices = fileReader.loadVerticesFromFile();
+        if (vertices != null) {
+            texture = new float[vertices.length / 3 * 2];
+            for (int i = 0; i < texture.length; i += 2) {
+                texture[i] = 0;    // Значение u
+                texture[i + 1] = 0; // Значение v
+            }
+            System.out.println("Vertices loaded successfully.");
+        }
+    }
+
+    private void loadFaces() {
+        faces = fileReader.loadFacesFromFile();
+        if (faces != null) {
```

Comment:
can we move this check and log to loadFacesFromFile?

Reply:
"We can do it, but I think it’s logical to handle it here since we need to perform the action ```faces = fileReader.loadFacesFromFile();``` in a separate function anyway."